tiatoolbox/models/architecture/utils.py

```diff
@@ -2,10 +2,60 @@
 
 from __future__ import annotations
 
+import sys
+from typing import Callable
+
 import numpy as np
 import torch
 from torch import nn
 
+from tiatoolbox import logger
+
+
+def compile_model(
+    model: nn.Module | None = None,
+    *,
+    mode: str = "default",
+    disable: bool = False,
+) -> Callable:
+    """A decorator to compile a model using torch-compile.
+
+    Args:
+        model (torch.nn.Module):
+            Model to be compiled.
+        mode (str):
+            Mode to be used for torch-compile. Available modes are
+            `default`, `reduce-overhead`, `max-autotune`, and
+            `max-autotune-no-cudagraphs`.
+        disable (bool):
+            If True, torch-compile will be disabled.
+
+    Returns:
+        Callable:
+            Compiled model.
+
+    """
+    if disable:
```

Comment:
I do not think we need this variable. I think we should only call this function if `not disable`